velox/functions/sparksql/window/tests/StreamingWindowTest.cpp

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/lib/window/tests/WindowTestBase.h"
+#include "velox/functions/prestosql/window/WindowFunctionsRegistration.h"
+#include "velox/functions/sparksql/window/WindowFunctionsRegistration.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+static const std::vector<std::string> kWindowFunctions = {
+    std::string("rank()"),
+    std::string("dense_rank()"),
+    std::string("percent_rank()"),
+    std::string("cume_dist()"),
+    std::string("row_number()"),
+    std::string("ntile(1)"),
+    std::string("sum(c2)"),
+    std::string("min(c2)"),
+    std::string("max(c2)"),
+    std::string("count(c2)"),
+    std::string("avg(c2)"),
+    std::string("sum(1)")};
+
+// The StreamingWindowTestBase class is used to instantiate parameterized window
+// function tests. The parameters are based on the function being tested
+// and a specific over clause. The window function is tested for the over
+// clause and all combinations of frame clauses. Doing so amortizes the
+// input vector and DuckDB table construction once across all the frame clauses
+// for a (function, over clause) combination.
+struct StreamingWindowTestParam {
+  const std::string function;
+  const std::string overClause;
+  const std::string frameClause;
+};
+
+class StreamingWindowTestBase : public WindowTestBase {
```

Comment:
For testing, consider changing WindowTestBase::testWindowFunction to run each query twice: first as is, second using streaming window over sorted inputs. Also, consider adding similar logic to the Fuzzer.

See AggregationTestBase::testAggregations( L628 in velox/functions/lib/aggregates/tests/AggregationTestBase.cpp

Reply:
@mbasmanova : WindowTestBase::testWindowFunction is used almost across all the tests. Adding streaming window here would significantly increase test time. I feel we should keep the separation at this level. wdyt ?

Its fine if we want to do both variations in the fuzzer though.